couchbase/content.md

```diff
@@ -0,0 +1,258 @@
+# What is Couchbase Server?
+
+[Couchbase Server](http://en.wikipedia.org/wiki/Couchbase_Server) is an open-source, distributed (shared-nothing architecture) NoSQL document-oriented database and key-value store that is optimized for interactive applications.
+
+Licensing information is covered towards the end of this guide.
+
+For support, please visit the [Couchbase support forum](https://forums.couchbase.com/) or `#couchbase` on irc.freenode.net.
+
+%%LOGO%%
+
+# How to use this image: QuickStart
+
+```
+docker run -d -p 8091:8091 couchbase/server
+```
+
+At this point go to http://localhost:8091 from the host machine to see the Admin Console web UI.  More details and screenshots are given below in the **Single host, single container** section.
+
+# Background Information
+
+## Networking
+
+Couchbase Server communicates on a number of different ports (see the [Couchbase Server documentation](http://docs.couchbase.com/admin/admin/Install/install-networkPorts.html "Network ports page on Couchbase Server documentation")). It also is not generally supported for nodes in a cluster to be behind any kind of NAT.  For these reasons, Docker's default networking configuration is not ideally suited to Couchbase Server deployments.
+
+There are several deployment scenarios which this Docker image can easily support. These will be detailed below, along with recommended network arrangements for each.
+
+## Volumes
+
+A Couchbase Server Docker container will write all persistent and node-specific data under the directory `/opt/couchbase/var`. We recommend mapping this directory to a directory on the host filesystem (using the `-v` option to `docker run`) for the following reasons:
+
+* **Persistence** Storing `/opt/couchbase/var` outside the container allows you to delete the container and re-create it later. You can even update to a container running a later point release of Couchbase Server without losing your data.
+* **Performance** In a standard Docker environment using a union filesystem, leaving `/opt/couchbase/var` "inside" the container will result in some amount of performance degradation.
```

Comment:
Since this directory is declared as a `VOLUME` in the `Dockerfile`, it will actually be automounted to a bind mount on the host by docker, just not in a place easily known to the user. But a user could just use `volumes-from` to migrate:
```console
$ docker run -d --name my-couchbase couchbase
$ # use database
$ docker stop my-couchbase
$ docker run -d --name new-couchbase --volumes-from my-couchbase couchbase
$ docker rm my-couchbase
```